Web.Client/Pages/Electives/HomeIndexMyElectives.razor.cs

```diff
@@ -18,8 +25,23 @@ protected override async Task OnInitializedAsync()
 		{
 			return;
 		}
+		await SubjectDataStore.EnsureDataAsync();
 
 		studentsProgress = await SubjectRegistrationProgressValidationFacade.GetProgressOfCurrentStudentAsync();
+
+		subjectIds = await StudentSubjectRegistrationFacade.GetAllRegistrationsOfCurrentStudentAsync();
+		foreach (var subjectId in subjectIds)
+		{
+			subjects.Add(await SubjectDataStore.GetByKeyAsync(subjectId.SubjectId.Value));
+			for (int i = 0; i < subjects.Count() - 1; i++)
+			{
+				if (subjects[i].ScheduleDayOfWeek == subjects[subjects.Count() - 1].ScheduleDayOfWeek && subjects[i].ScheduleSlotInDay == subjects[subjects.Count() - 1].ScheduleSlotInDay)
+				{
+					subjectsCollide = true;
+				}
+			}
```

Comment:
spíše bych si to z data-store vytáhl do lokální proměnné `subject` a s tou pak i pracoval v cyklu. Není moc přehledné, pokud tam používáš `subjects[subject.Count() - 1]` (i kdybys to zapsal jako `subjects[^1]` nebo `subjects.Last()`).
Podle mě by bylo o dost přehlednější to přeskupit nějak takto:
```csharp
var subject = await SubjectDataStore.GetByKeyAsync(subjectId.SubjectId.Value);
foreach (var otherSubject in subjects)
{
  if ((subject.ScheduleDayOfWeek == otherSubject.ScheduleDayOfWeek)
       && (subject.ScheduleSlotInDay == otherSubject.ScheduleSlotInDay))
  {
      subjectsCollide = true;
  }
}
subjects.Add(subject);

```